### PR TITLE
Single common `fn.Client` factory for all commands

### PR DESCRIPTION
**Pull request by matejvasek**
Added single common factory function of `fn.Client` for all commands.
This allows to set all important settings at one place.

### CODE REVIEW

**Comment by matejvasek** [comment]
@lkingland this is how it could look like.

File: cmd/client.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package cmd
```
Comment (matejvasek) [inline cmd/client.go:1]: @lkingland here is the factory

**Review by lance — COMMENTED** [review]
Overall this lgtm and seems to be a big improvement over the more ad hoc method of instantiating a client within each subcommand. I do have one small question about some interrupt behavior that seems to have been removed. I also have some lingering concern about how we will address additional build strategies (e.g. local or s2i). I wouldn't expect you to have given a lot of thought to that, as it's only recently been discussed. But if you have, I would be interested in how you see that working with this flow.

File: cmd/deploy.go [hunk 2]
```diff
@@ -210,17 +165,7 @@ func runDeploy(cmd *cobra.Command, _ []string, clientFn deployClientFn) (err err
...
-	defer rt.Close()
-	config.Transport = rt
-
-	client, err := clientFn(config)
-	if err != nil {
-		if err == terminal.InterruptErr {
-			return nil
-		}
-		return err
-	}
```
[the remark this answers is not in the document]
Comment (matejvasek) [reply cmd/deploy.go:223]: Well the factory now doesn't return error, so there is nothing to check. Also I doubt that it ever returned `terminal.InterruptErr`. That error makes sense for the prompt function e.g. 
```go
config, err := newBuildConfig().Prompt()
	if err != nil {
		if err == terminal.InterruptErr {
			return nil
		}
		return
	}
```

[the remark this answers is not in the document]
Comment (matejvasek) [reply cmd/deploy.go:223]: It was added in
https://github.com/knative-sandbox/kn-plugin-func/commit/7a24a103efeebc6542933409545bdc3709e0da2e#diff-e61f952a089e5960dd93ef2a48a83b1187c9700efabc40974fb9f2d9b50b4a7eR128
But the `Pusher` constructor itself shouldn't return the interrupt error.
Subsequent `Push()` calls on in could however.

[the remark this answers is not in the document]
Comment (matejvasek) [reply cmd/deploy.go:223]: So the check was unnecessary all along.

File: cmd/build.go [hunk 3]
```diff
@@ -220,8 +181,6 @@ type buildConfig struct {
 	// with interactive prompting (only applicable when attached to a TTY).
 	Confirm bool
 	Builder string
-
-	Transport http.RoundTripper
```
[the remark this answers is not in the document]
Comment (matejvasek) [reply cmd/build.go:224]: Does the delete or the info command invoke some cluster internal service?
If not I don't think it's necessary.

**Comment by matejvasek** [comment]
>I also have some lingering concern about how we will address additional build strategies (e.g. local or s2i). I wouldn't expect you to have given a lot of thought to that, as it's only recently been discussed.

Indeed I didn't think much about it, but I think these changes shouldn't affect it much.

**Pull request merged**